### PR TITLE
BookService have been added to call against the BookController endpoints

### PR DESCRIPTION
**Pull request by jeongoks**
Part of #76 

- [+] Added the `BookService` to call the `BooksController` endpoints.
- [+]  Added the `BookService` to the `PubHubApiService`.
- [~] Modified that `UserService` to call the right endpoints, as they're now using an `userId` in the parameter of `UsersController`.

### CODE REVIEW

File: PubHub/PubHub.Common/ApiService/PubHubApiService.cs [hunk 1]
```diff
@@ -17,13 +17,15 @@ public class PubHubApiService
         public string ClientName { get; }
 
         public UserService Users { get; }
+        public BookService Books { get; set; }
```
Comment (Sjimooon) [inline PubHub/PubHub.Common/ApiService/PubHubApiService.cs:20]: Is `set` nedded?

**Comment by ZhakalenDk** [comment]
I know I've done the centralized Unit of Work ani-pattern in my MBudget project.
But don't tink we should use it here.

That is; your [PubHubApiService](https://github.com/Mike-Mortensen-Portfolio/PubHub_H6_Final/blob/ad9c253019b9ac1a0077b6fa559e5b9b162436c0/PubHub/PubHub.Common/ApiService/PubHubApiService.cs) is a so called Unit of Work, which is an anti-pattern. It's much better to seperate the services and only inject the ones that a needed in th ecurretn context (:

**Review by Sjimooon — APPROVED** [review]
🏄‍♂️

**Comment by jeongoks** [comment]
> I know I've done the centralized Unit of Work ani-pattern in my MBudget project. But don't tink we should use it here.
> 
> That is; your [PubHubApiService](https://github.com/Mike-Mortensen-Portfolio/PubHub_H6_Final/blob/ad9c253019b9ac1a0077b6fa559e5b9b162436c0/PubHub/PubHub.Common/ApiService/PubHubApiService.cs) is a so called Unit of Work, which is an anti-pattern. It's much better to seperate the services and only inject the ones that a needed in th ecurretn context (:

So you want to move below out from the `PubHubApiService`? I will need more time, again, to try to work around this then. So unless you know how to do it quickly... 🤔 i think that'd be better so that we can move on with other issues as well.

**Comment by ZhakalenDk** [comment]
> > I know I've done the centralized Unit of Work ani-pattern in my MBudget project. But don't tink we should use it here.
> > That is; your [PubHubApiService](https://github.com/Mike-Mortensen-Portfolio/PubHub_H6_Final/blob/ad9c253019b9ac1a0077b6fa559e5b9b162436c0/PubHub/PubHub.Common/ApiService/PubHubApiService.cs) is a so called Unit of Work, which is an anti-pattern. It's much better to seperate the services and only inject the ones that a needed in th ecurretn context (:
> 
> So you want to move below out from the `PubHubApiService`? I will need more time, again, to try to work around this then. So unless you know how to do it quickly... 🤔 i think that'd be better so that we can move on with other issues as well.

It's an easy change.
All you essentially have to do is delete the UoW class and have each service instantiate their own reference of the `HTTPClient`.
Basically just make a service that can be DP injected where it's needed instead of having a large UoW injected and instantiate every single service even though you're just using one of them (:

**Review by ZhakalenDk — APPROVED** [review]

**Pull request merged**